executor/builder.go

```diff
@@ -122,6 +122,7 @@ func newExecutorBuilder(ctx sessionctx.Context, is infoschema.InfoSchema, ti *Te
 		ctx:              ctx,
 		is:               is,
 		Ti:               ti,
+		snapshotTSCached: isStaleness,
```

Comment:
I didn't understand. Where is the `snapshotTS` set in this case?

Reply:
for stale read `snapshotTS` is set when `executorBuilder` is constructed

Reply:
just under this line